Adyen/UI/Form/Items/Address/FormFullAddressItem.swift

```diff
@@ -64,76 +62,98 @@ public final class FullFormAddressItem: FormValueItem<AddressInfo, AddressStyle>
+            item.validator = LengthValidator(minimumLength: 2, maximumLength: 70)
+        }
 
-    internal lazy var provinceTextItem: FormTextInputItem = {
-        let cityTextItem = FormTextInputItem(style: style.textField)
-        cityTextItem.title = localizedString(.provinceOrTerritoryFieldTitle, localizationParameters)
-        cityTextItem.placeholder = localizedString(.provinceOrTerritoryFieldPlaceholder, localizationParameters)
-        cityTextItem.validator = LengthValidator(minimumLength: 2, maximumLength: 50)
-        cityTextItem.validationFailureMessage = validationMessage
-        cityTextItem.identifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: "provinceOrTerritory")
+        bind(item: item, to: field)
+        return item
+    }
 
-        return cityTextItem
-    }()
+    private func bind(item: FormRegionPickerItem, to field: AddressFields) {
+        observers[field].map { remove($0) }
+        observers[field] = bind(item.publisher, at: \.identifier, to: self, at: \.value.stateOrProvince)
+    }
 
-    internal lazy var postalCodeTextItem: FormTextInputItem = {
-        let cityTextItem = FormTextInputItem(style: style.textField)
-        cityTextItem.title = localizedString(.postalCodeFieldTitle, localizationParameters)
-        cityTextItem.placeholder = localizedString(.postalCodeFieldPlaceholder, localizationParameters)
-        cityTextItem.validator = LengthValidator(minimumLength: 2, maximumLength: 30)
-        cityTextItem.validationFailureMessage = validationMessage
-        cityTextItem.identifier = ViewIdentifierBuilder.build(scopeInstance: self, postfix: "postalCode")
-        return cityTextItem
-    }()
+    private func bind(item: FormTextInputItem, to field: AddressFields) {
+        observers[field].map { remove($0) }
+
+        switch field {
+        case .street:
+            observers[.street] = bind(item.publisher, to: self, at: \.value.street)
+        case .houseNumberOrName:
+            observers[.houseNumberOrName] = bind(item.publisher, to: self, at: \.value.houseNumberOrName)
+        case .appartment:
+            observers[.appartment] = bind(item.publisher, to: self, at: \.value.apartment)
+        case .postalCode:
+            observers[.postalCode] = bind(item.publisher, to: self, at: \.value.postalCode)
+        case .city:
+            observers[.city] = bind(item.publisher, to: self, at: \.value.city)
+        case .stateOrProvince:
+            observers[.stateOrProvince] = bind(item.publisher, to: self, at: \.value.stateOrProvince)
+        }
+    }
 
     /// :nodoc:
     override public func build(with builder: FormItemViewBuilder) -> AnyFormItemView {
         builder.build(with: self)
     }
 
 }
+
+private func localizedString(_ key: LocalizationKey?, _ parameters: LocalizationParameters?, _ arguments: CVarArg...) -> String? {
+    guard key != nil else { return nil }
+    return localizedString(key.unsafelyUnwrapped, parameters, arguments) as String
```

Comment:
why do we need `unsafelyUnwrapped`? why not like this?:

```
guard let key = key else { return nil }
     return localizedString(key, parameters, arguments) as String
```

Reply:
because `key ` is enum and you can't do `guard let key = key` on enums. It have to be unwrapped to be checked, else enum is only checked as `guard case let key = .KEY`

Reply:
i think that's a perfect place to use map instead of this function
```
item.title = viewModel.labels[.stateOrProvince]
            .map { localizedString($0, localizationParameters) }
```

Reply:
👍